feat: onboarding global dashboard, multiple onboarding sequence

cdrage opened the pull request:
feat: onboarding global dashboard, multiple onboarding sequence

### What does this PR do?

* Adds a "global onboarding" dashboard to the welcome page
* Modifies Onboarding.svelte to allow being able to run multiple
  onboarding sequences.
* When multiple onboarding sequences are to be completed, onboarding
  goes "full screen" and hides the status bar / side bar.

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

https://github.com/containers/podman-desktop/assets/6422176/f7ee16ee-fd0e-40d9-b11c-7c9f3ae70070

### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop/issues/5162
Closes https://github.com/containers/podman-desktop/issues/5161
Closes https://github.com/containers/podman-desktop/issues/5163

### How to test this PR?

1. Remove `welcome.version` from ~/.local/share/containers/podman-desktop/configuration/settings.json
so you can access the welcome screen
2. Click on multiple onboarding selections
3. Go through the onboarding sequence

<!-- Please explain steps to reproduce -->

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

cdrage commented:
![Screenshot 2023-12-14 at 3 35 24 PM](https://github.com/containers/podman-desktop/assets/6422176/797b0c44-da88-4d3d-9558-1098568fc2f8)

I'd like some suggestions on how we could improve showing the onboarding. Right now I am showing anything that has onboarding enabled, not necessarily "needing" setting up.

This also would be removing the "featured" list..

So would we add featured items as well and simply not have the checkbox available?

@mairin @deboer-tim

cdrage commented:
TODO:

- [ ] Add tests
- [ ] Finalize UI

benoitf commented:
hello, shouldn't the checkboxes be purple and smaller like the one used for telemetry just below the cards ?

and is that the default should be checked ?

cdrage commented:
> hello, shouldn't the checkboxes be purple and smaller like the one used for telemetry just below the cards ?
> 
> and is that the default should be checked ?

![288557538-fe7aca3f-e998-4102-a76a-fccc073c8ccf (1)](https://github.com/containers/podman-desktop/assets/6422176/f95c8acf-4252-4345-8449-044ef66b1646)

I'm going by the mockup so that's why they are larger and not clicked by default.

My UI & integration questions are:

Should I be removing the "featured list" as now we are using onboarding?

In the mockup is shows "featured" labels but you can check mark them even though it's not onboarding, what does that mean?

Should we be showing ALL the onboarding even if it doesn't require setting up? Or else it would be very empty vs the featured list. Each onboarding process has a "summary" page at the end, so we would show that? Or do we just mark it as grayed out / already setup?

mairin commented:
> I'm going by the mockup so that's why they are larger and not clicked by default.

👍🏻 

> Should I be removing the "featured list" as now we are using onboarding? 
> In the mockup is shows "featured" labels but you can check mark them even though it's not onboarding, what does that mean?

Ah this is a good question. The mockup was created with the assumption every extension shown has an onboarding flow. This is clearly not the case now. So I would only show the extensions that have an onboarding flow available and drop the featured list. (We show the featured list on the dashboard anyway, so they get some good promotion to users without being in the welcome screen.)

> Should we be showing ALL the onboarding even if it doesn't require setting up? Or else it would be very empty vs the featured list. Each onboarding process has a "summary" page at the end, so we would show that? Or do we just mark it as grayed out / already setup?

So for Linux users this question is of particular interest wrt Podman... because if they're using a RHT family distro anyway, they'll have podman already installed. But they may not have set up configuration specific to Podman Desktop in respect to that - I guess the main one is does it auto start or not. So my initial thoughts are:

- I cannot think of any cases where the user going through the welcome screen - which should only appear on the first run of the application - would actually have the extension + software already installed except for: (1) Podman on Linux (2) manually installing podman-machine on Mac/Win (3) manually installing crc on mac/win/linux

- For the Linux case, I'd want to see the podman extension options even if I already have podman installed, because I want to see what preferences I can set wrt it.

- For the manually-installed podman-machine / crc... I think going through the flow lets me see whether or not it detected my manually installed thing and tweak configuration, so it's not a problem to go through it again.

So I would err on the side of letting them go through it again. I think if we detect an installation, we could flag that with one of those little taggy things at the bottom of the card, "Installation detected" or something like that. There may even be a mockup for this in the penpot file, I'll dig around and look, because it feels familiar.

mairin commented:
> So I would err on the side of letting them go through it again. I think if we detect an installation, we could flag that with one of those little taggy things at the bottom of the card, "Installation detected" or something like that. There may even be a mockup for this in the penpot file, I'll dig around and look, because it feels familiar.

Ah yeh I went back through the mockup file. The "detected" badge is meant to show we detected that you had the thing installed.

mairin commented:
Will talk about this at the UX call tomorrow!

mairin commented:
Quick notes from the UX call yesterday:

- Keep the featured extensions at the bottom, but use the small / non-clickable version. Maybe drop the button-looking background off of them so they don't look like clickable buttons since they're not. 
- Have some logic in the featured extensions area to de-duplicate what's offered via onboarding checkbox options at the top.
- Remove the podman option if on linux for the featured extension area.

cdrage commented:
Ready for review!

Changes:
- Updated UI to still show featured list, but remove the ones that already have onboarding
- Added a *a lot* of tests

https://github.com/containers/podman-desktop/assets/6422176/39118340-ad3d-45ef-b0fe-3231d1dece3c

cdrage commented:
@benoitf @dgolovin @jeffmaury Ready for review 🎉

odockal commented:
@cdrage Please also map this PR or issue addresses here into this [epic](https://github.com/containers/podman-desktop/issues/5491), to be sure we are counting in QE with this feature to be covered by the tests in the future.

cdrage commented:
> Unfortunately it doesn't work for me. The podman onboarding is broken, even if i start it from the resources page the page is empty - no errors in the console :/ (also tried to run it from the main and works as expected, so i guess it is some update from here). I'm using Windows.
> 
> EDIT: to be precise, during the global onboarding, the podman onboarding get stuck at the installing podman step (there should be a dialog asking for a confirmation to install it and it does not appear). When i start the podman onboarding from the resources page, i don't see anything. All black
> 
> Rgearding the UI, not sure if it's correct but i see a light grey row between the top toolbar and the onboarding header. Is that expected?
> 
> ![image](https://private-user-images.githubusercontent.com/49404737/305101436-eec8a2b5-fa9b-485a-894c-20468e65a84f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwMDg0MzMsIm5iZiI6MTcwODAwODEzMywicGF0aCI6Ii80OTQwNDczNy8zMDUxMDE0MzYtZWVjOGEyYjUtZmE5Yi00ODVhLTg5NGMtMjA0NjhlNjVhODRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE1VDE0NDIxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjYzA3MzI2ZDEwYWQ3OTcxN2Q0YWNlMjczNTEwZmQ5MTUyMjUzMmE1ZGE1YTAxNDNhZTNkNjk0MDRmMTAwOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.W8_dpd1il39Aqg6coRzhGS5NBmxeO7eEMMm_rknHH2I)

Any way to take a video? Just trying to reproduce it and have been unable to (even uninstalling podman and trying it again with a fresh onboarding).

cdrage commented:
> Unfortunately it doesn't work for me. The podman onboarding is broken, even if i start it from the resources page the page is empty - no errors in the console :/ (also tried to run it from the main and works as expected, so i guess it is some update from here). I'm using Windows.
> 
> EDIT: to be precise, during the global onboarding, the podman onboarding get stuck at the installing podman step (there should be a dialog asking for a confirmation to install it and it does not appear). When i start the podman onboarding from the resources page, i don't see anything. All black
> 
> Rgearding the UI, not sure if it's correct but i see a light grey row between the top toolbar and the onboarding header. Is that expected?
> 
> ![image](https://private-user-images.githubusercontent.com/49404737/305101436-eec8a2b5-fa9b-485a-894c-20468e65a84f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgwOTMxNjYsIm5iZiI6MTcwODA5Mjg2NiwicGF0aCI6Ii80OTQwNDczNy8zMDUxMDE0MzYtZWVjOGEyYjUtZmE5Yi00ODVhLTg5NGMtMjA0NjhlNjVhODRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE2VDE0MTQyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkZWI4NjFkMTEzMDMzZGJlZDliZGFiNWVhNjVlMjU4NDNhMDU5N2VmYjk1Y2I0MzJjMDE2MjkyODU4ZTVjMjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6t6fPU4OuK1svezoieJRUdu3AgYCMzdB1Ud6Q0ZlSLI)

I've just rebased against master and I'm trying to reproduce it on Windows and unfortunately I'm unable to get the black screen.

cdrage commented:
https://github.com/containers/podman-desktop/assets/6422176/dbd00a66-37b0-449e-a962-21dc5f007f4b

@lstocchi Indeed it was a z-value error! It had me scratching my head why it wasn't appearing, and it was just getting hidden.

I've updated the PR with the small minor change and it's ready for a review.

@lstocchi @jeffmaury @dgolovin

lstocchi commented on packages/renderer/src/lib/welcome/WelcomePage.svelte:
```diff
@@ -3,32 +3,61 @@ import bgImage from './background.png';
+        selected: false,
+      };
+    });
+    console.log(onboardingProviders);
```
to be removed?

lstocchi reviewed (commented):
I still see a black page when i skip the global onboarding and i select the podman one from the resources page (or compose).

After investigating a bit, the problem is in this function
https://github.com/containers/podman-desktop/blob/46d8ea646eb0eabf266a74bc16b046786d668cf1/packages/renderer/src/lib/onboarding/onboarding-utils.ts#L44-L57

When the user skips the global onboarding you set all steps to "skipped" in the right way. However, by doing this, the method above will set the entire onboarding to complete when the status of the last step gets updated (to skipped). 
We do not do any check here, so no matter which status the last step has. If the last step gets updated we mark the onboarding as completed. For this reason, if you then try to open the onboarding from the resources page you see nothing bc for desktop the onboarding has been completed before.

To avoid it you just need to add a check before updating the entire onboarding status like 

`if (lastCompletedStep?.id === step.id && status === STATUS_COMPLETED) {`

this way the onboarding will be marked as complete only if the last step is completed.

lstocchi commented on packages/renderer/src/lib/onboarding/Onboarding.svelte:
```diff
@@ -71,7 +73,7 @@ let executedCommands: string[] = [];
 
 /*
-$: enableNextButton = false;*/
+  $: enableNextButton = false;*/
```
to be removed?

lstocchi reviewed (approved):
LGTM. Thanks @cdrage , great job 🚀

jeffmaury commented:
If I don't have Compose installed (for example) and I go to resource pages, it tells me Set up.
Can't we have the Compose card selected in that case in the onboard screen ?

cdrage commented:
> If I don't have Compose installed (for example) and I go to resource pages, it tells me Set up. Can't we have the Compose card selected in that case in the onboard screen ?

That is intentional with the design not to have it check-boxed as users may not want to initially set up the extension.

I also kept encountering this bug: https://github.com/containers/podman-desktop/issues/5249 which would make loading context on the initial welcome screen not very stable.

slemeur commented:
> That is intentional with the design not to have it check-boxed as users may not want to initially set up the extension.

If it's not setup, I think we should default its installation / configuration. Those who don't want it, can uncheck the box.
The users who already have it installed, will not need this step as the compatibility mode will cover it. (in this case, it should be skipped)
We still do have quite a few users who are reporting struggling with the configuration of compose - hopefully the onboarding sequence will make things much clearer and simpler for the end-users

cdrage commented:
> > That is intentional with the design not to have it check-boxed as users may not want to initially set up the extension.
> 
> If it's not setup, I think we should default its installation / configuration. Those who don't want it, can uncheck the box. The users who already have it installed, will not need this step as the compatibility mode will cover it. (in this case, it should be skipped) We still do have quite a few users who are reporting struggling with the configuration of compose - hopefully the onboarding sequence will make things much clearer and simpler for the end-users

Managed to get this to work!

I've updated the PR to include retrieving the current providers (in order to get the status), as well as the global context and pre-select the ones that have enablement.

I've added it as a separate PR and implemented the requested changes @slemeur 


https://github.com/containers/podman-desktop/assets/6422176/098a601d-8d25-4b16-b2f9-c3f896365c57

@lstocchi do you mind testing this again when you get the chance? As you already have a Windows setup all ready to go 🎉

lstocchi commented:
Tested again, same as Jeff

cdrage commented:
@jeffmaury @lstocchi Looks like I forgot to add the overflow-y-auto.

Re-added it and now you can scroll correctly:

https://github.com/containers/podman-desktop/assets/6422176/0b24cce3-523f-4fcb-8e15-e817ecc8c56d


To answer your question on:

"Also if Podman is installed but no Podman machine I think we should have the Podman onboarding selected in the onboarding page"

We use the global context  / onbarding onWhen clauses and would have to implement provider-specific support, see how the code works with showing "Create new ..." vs "Setup ...": https://github.com/cdrage/podman-desktop/blob/3d81f974ffef98015aa8957073815a103ca4305b/packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte#L402

As this PR is getting too big, let's do it in a follow up PR?

jeffmaury commented:
The Podman onboarding is broken for me in the autostart page

![image](https://github.com/containers/podman-desktop/assets/695993/94803364-6828-4e88-83f4-9fc52b6f7570)

jeffmaury reviewed (approved):
LGTM

benoitf reviewed (changes requested):
The first screen people will see is broken 
![image](https://github.com/containers/podman-desktop/assets/436777/7c946099-48a3-4729-9745-beab13cbadfe)

we had in the past something 'slick' here we have a vertical scrollbar and I don't see the Podman Desktop logo correctly
Also we don't see the line
![image](https://github.com/containers/podman-desktop/assets/436777/34c97dec-bbfb-418a-ab4b-bb06f1eb07f4)
without scrolling

IMHO this first screen should be displayed without any scrolling and we should see every item. This is the first thing users will see


Then, why the checkboxes are not checked by default for the onboarding ?

Clicking on a checkbox, it's moving the UI when it's checked
https://github.com/containers/podman-desktop/assets/436777/dd718e8a-b425-456f-b148-8a16b0a44e96
Not sure why the checkboxes are blue and not purple ?

On the UI, I have the feeling that I have a toggle to switch from one step to another step
![image](https://github.com/containers/podman-desktop/assets/436777/cefaa3f7-7a56-4c70-99c4-b8695f1e93d6)

I think I could click here to go back to previous steps




For Podman onboarding, I see the screen
![image](https://github.com/containers/podman-desktop/assets/436777/8cd6d8ed-b630-4b82-bfd6-68ae4aa163cd)

so I have the feeling that everything is installed but if I click next I have
![image](https://github.com/containers/podman-desktop/assets/436777/fd4ed09c-0739-4b02-8504-d1c90e00203a)

I don't see what this screen is bringing I would expected to not see this screen as it was saying that it was already installed on the previous screen.

cdrage commented:
> The first screen people will see is broken ![image](https://private-user-images.githubusercontent.com/436777/309190198-7c946099-48a3-4729-9745-beab13cbadfe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODIzNjMsIm5iZiI6MTcwOTU4MjA2MywicGF0aCI6Ii80MzY3NzcvMzA5MTkwMTk4LTdjOTQ2MDk5LTQ4YTMtNDcyOS05NzQ1LWJlYWIxM2NiYWRmZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQxOTU0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MzM2MTM1NzkwYTUwZTljYTgzMTRiNzZhNTJiY2YwNTU1ZjY4YmU4YmIzNmJjMzZmNjQ0ODI4ZjVmYjI2YzZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TZxXbfk9_lOfVj0iJwKN-dR6SaT_dpvzQgpN-8Akf6M)
> 
> we had in the past something 'slick' here we have a vertical scrollbar and I don't see the Podman Desktop logo correctly Also we don't see the line ![image](https://private-user-images.githubusercontent.com/436777/309192102-34c97dec-bbfb-418a-ab4b-bb06f1eb07f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODIzNjMsIm5iZiI6MTcwOTU4MjA2MywicGF0aCI6Ii80MzY3NzcvMzA5MTkyMTAyLTM0Yzk3ZGVjLWJiZmItNDE4YS1hYjRiLWJiMDZmMWViMDdmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQxOTU0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZGEyMGRmMmNjM2Q5ODA1MDQzNjRmOTYxMjc5MTE4MGUwZTdhZmE3YTU0ZWVmZGIwYmJjMWNiMWQ3MmI0NjI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UL3UdBrgo4QpF7BCB5KmbmCGjdik8nwy9Tv6j820tE4) without scrolling
> 
> IMHO this first screen should be displayed without any scrolling and we should see every item. This is the first thing users will see
> 
> Then, why the checkboxes are not checked by default for the onboarding ?
> 
> Clicking on a checkbox, it's moving the UI when it's checked https://github.com/containers/podman-desktop/assets/436777/dd718e8a-b425-456f-b148-8a16b0a44e96 Not sure why the checkboxes are blue and not purple ?
> 
> On the UI, I have the feeling that I have a toggle to switch from one step to another step ![image](https://private-user-images.githubusercontent.com/436777/309191387-cefaa3f7-7a56-4c70-99c4-b8695f1e93d6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODIzNjMsIm5iZiI6MTcwOTU4MjA2MywicGF0aCI6Ii80MzY3NzcvMzA5MTkxMzg3LWNlZmFhM2Y3LTdhNTYtNGM3MC05OWM0LWI4Njk1ZjFlOTNkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQxOTU0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NmI4YjVmYTY0OWI1NmNjOGY3NzQ0OWYyZTlmOTM3ODFhMzY1NDFjNmYwNzY5OGE5NmEwYzc3MjcwNjliNzQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2a-GDSZiHXQTJT24dJlllv9tMraqV_r-_nqdUI0nXbM)
> 
> I think I could click here to go back to previous steps
> 
> For Podman onboarding, I see the screen ![image](https://private-user-images.githubusercontent.com/436777/309192864-8cd6d8ed-b630-4b82-bfd6-68ae4aa163cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODIzNjMsIm5iZiI6MTcwOTU4MjA2MywicGF0aCI6Ii80MzY3NzcvMzA5MTkyODY0LThjZDZkOGVkLWI2MzAtNGI4Mi1iZmQ2LTY4YWU0YWExNjNjZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQxOTU0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yOTUzYWMwZGE1ZDFlODE1OTdkMDQwZTQ0ODY1ZGM3MGQyNDhiNjQyMWFmYjVmY2U2YjFmOGEzN2U1OGFjMjY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ETurZMwrGSI5kLgmLnh4MMOPfHXtuxSYLWYBa2lp6r0)
> 
> so I have the feeling that everything is installed but if I click next I have ![image](https://private-user-images.githubusercontent.com/436777/309192979-fd4ed09c-0739-4b02-8504-d1c90e00203a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk1ODIzNjMsIm5iZiI6MTcwOTU4MjA2MywicGF0aCI6Ii80MzY3NzcvMzA5MTkyOTc5LWZkNGVkMDljLTA3MzktNGIwMi04NTA0LWQxYzkwZTAwMjAzYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNFQxOTU0MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OTkzMTdmYWFiOTZhZmRhMzAwN2NhYzdiNTRmMTgxOGE0Yzc5YTRiNDVlOTE3MjY5MWIzYzJhNGZjZDI3NjcxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.eUn1J_NQ7lHrbIRrsx0ulaqx_Cu_hQ98K3Ze8Alg5mQ)
> 
> I don't see what this screen is bringing I would expected to not see this screen as it was saying that it was already installed on the previous screen.

![Screenshot 2024-03-04 at 3 07 48 PM](https://github.com/containers/podman-desktop/assets/6422176/ee5e669c-85c2-437b-b641-4ee85b5a6069)

With #6268 we got some space back now.

The above implementation of dashboard was done via the UI mockup.

To answer your questions:

1. The purple "dots" are from the UI mockup. I'm unsure what we should do with regards to that / make it better, as they do look "clickable"
2. Also unsure regarding creating space, as we concluded we needed the "featured" section too, should we just remove the "Configure these and more under Settings" text?
3. Checkboxes are only checked if it's needing Setup, similar to the "Setup ..." button appearing under Extensions in settings. However, you CAN click on it again to go through the process such as the "extra" options like toggling auto start / on and off.

benoitf commented:
@cdrage there was agreement to get rid of featured extensions on the welcome page (there are on the extensions page and welcome page now)

it'll be make the screen easier (so no need to change something on the layout)

cdrage commented:
Updated based on review, this removes portions of the "featured extensions" tests as well as they are no longer applicable / wont be showed.

The new screen looks like this:
![Screenshot 2024-03-06 at 2 13 48 PM](https://github.com/containers/podman-desktop/assets/6422176/078921a6-2562-4292-b3d7-f372c0e34a32)

Other PR's such as the compose onboarding screen is located here: https://github.com/containers/podman-desktop/pull/6270

benoitf commented:
unit tests are failing

cdrage commented:
> unit tests are failing

Was just linter, had to remove some tests since featured is no longer there, re-running now!

benoitf commented:
rebasing commits on top of main branch

benoitf commented:
@cdrage I tried, I'm still wondering why the checkboxes are not checked by default
I expect to go through all onboarding steps on a first run (by default)

![image](https://github.com/containers/podman-desktop/assets/436777/b5a62112-41bc-4e84-84f7-08d471a5f867)


also as reported previously, clicking on a card to check the item is moving the screen to the top

https://github.com/containers/podman-desktop/assets/436777/bb5676da-d19c-42a1-9c2e-c94e83c1c46c

And then I don't understand why I'm facing two screens for Podman saying it's installed

I recorded the screens
for Compose and Kubectl it's working 'as expected' , I see quickly a 'Checking ...' screen redirecting to a 'is installed' but for Podman I've got like duplicated content : I see a first 'podman is installed' screen then I click on next and I get another 'is installed' screen. I would expect the same steps than for compose and kubectl. First a 'Checking...' redirecting to the 'is installed' screen

https://github.com/containers/podman-desktop/assets/436777/1c004455-4b5b-4171-ac4e-9fbef49e772f

cdrage commented:
As per call discussions, 

will remove context / when check code and instead have everything checkboxed.

cdrage commented:
Ready for another review. There have been multiple changes so please read carefully:

- The onboarding is automatically checked now
- Changes have been made to the "double installation" page and it has now been changed to "Podman settings" which is more fitting: https://github.com/containers/podman-desktop/pull/6308
- There is a rendering bug for onboarding: https://github.com/containers/podman-desktop/issues/6313 and I have added a temporary fix here: https://github.com/containers/podman-desktop/pull/6317/commits/6e42d34c6bc944e3a165b9c776b5b4f73a9eb687
- The button that changes the width issue is located here: https://github.com/containers/podman-desktop/issues/6304 this is because the other buttons have different border's, but the "default" one has no border and  that makes it slightly smaller. This may be a longer PR as we will have to make sure that between default / primary / secondary and danger, they are all the same width.  All of them are very slightly different widths due to the use of borders.
